From 9145b94e37d1ca1daf30840a7e613182d2fd4acd Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:12:24 -0400
Subject: [virt] kvm/ppc: book3s/pr - Export kvmppc_copy_to

Message-id: <1410765214-16377-9-git-send-email-dgibson@redhat.com>
Patchwork-id: 94764
O-Subject: [PATCH 08/78] KVM: PPC: Book3S: PR: Export kvmppc_copy_to|from_svcpu
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1123145

Prerequisite to backport 699a0ea0823d32030b0666b28ff8633960f7ffa7

Signed-off-by: David Gibson <dgibson@redhat.com>

commit c9dad7f9db4ed42de37d3f0ef2b2c0e10d5b6f92
Author: Alexander Graf <agraf@suse.de>
Date:   Fri Nov 29 02:27:23 2013 +0100

    KVM: PPC: Book3S: PR: Export kvmppc_copy_to|from_svcpu

    The kvmppc_copy_{to,from}_svcpu functions are publically visible,
    so we should also export them in a header for others C files to
    consume.

    So far we didn't need this because we only called it from asm code.
    The next patch will introduce a C caller.

    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/kvm_book3s.h b/arch/powerpc/include/asm/kvm_book3s.h
index 1e9c26f..83851aa 100644
--- a/arch/powerpc/include/asm/kvm_book3s.h
+++ b/arch/powerpc/include/asm/kvm_book3s.h
@@ -189,6 +189,10 @@ extern void kvmppc_hv_entry_trampoline(void);
 extern u32 kvmppc_alignment_dsisr(struct kvm_vcpu *vcpu, unsigned int inst);
 extern ulong kvmppc_alignment_dar(struct kvm_vcpu *vcpu, unsigned int inst);
 extern int kvmppc_h_pr(struct kvm_vcpu *vcpu, unsigned long cmd);
+extern void kvmppc_copy_to_svcpu(struct kvmppc_book3s_shadow_vcpu *svcpu,
+     struct kvm_vcpu *vcpu);
+extern void kvmppc_copy_from_svcpu(struct kvm_vcpu *vcpu,
+       struct kvmppc_book3s_shadow_vcpu *svcpu);
 
 static inline struct kvmppc_vcpu_book3s *to_book3s(struct kvm_vcpu *vcpu)
 {
-- 
1.7.1